lib/redis/index.ts

```diff
@@ -217,17 +222,17 @@ Redis.prototype.resetOfflineQueue = function () {
 };
 
 Redis.prototype.parseOptions = function () {
-  this.options = {};
+  this.options = { ...Redis.defaultOptions };
   let isTls = false;
   for (let i = 0; i < arguments.length; ++i) {
     const arg = arguments[i];
     if (arg === null || typeof arg === "undefined") {
       continue;
     }
     if (typeof arg === "object") {
-      defaults(this.options, arg);
+      Object.assign(this.options, arg);
```

Comment:
this.options is mutating, but you dont ensure, that the values in this.options are preserved

```suggestion
this.options = {...args, ...this.options};
```

Reply:
maybe your solution is actually correct. I would also expect that the parseURL is overwriting this.options.